src/browser/BrowserShared.cpp

```diff
@@ -35,10 +36,25 @@ namespace BrowserShared
         return QStandardPaths::writableLocation(QStandardPaths::RuntimeLocation) + "/app/" + "org.keepassxc.KeePassXC"
                + serverName;
 #elif defined(Q_OS_UNIX) && !defined(Q_OS_MACOS)
-        // Use XDG_RUNTIME_DIR instead of /tmp if it's available
-        QString path = QStandardPaths::writableLocation(QStandardPaths::RuntimeLocation);
-        return path.isEmpty() ? QStandardPaths::writableLocation(QStandardPaths::TempLocation) + serverName
-                              : path + serverName;
+        // Use XDG_RUNTIME_DIR if available, else use /tmp.
+        QString xdgRuntimePath = QStandardPaths::writableLocation(QStandardPaths::RuntimeLocation);
+        if (xdgRuntimePath.isEmpty())
+            return QStandardPaths::writableLocation(QStandardPaths::TempLocation) + serverName;
+
+        QDir xdgRuntimeDir = QDir(xdgRuntimePath);
+
+        // Put the socket in a dedicated directory.
+        // This directory will be easily mountable by sandbox containers.
+        QDir socketDir = xdgRuntimeDir.filePath("app/org.keepassxc.KeePassXC");
+        if (!socketDir.exists())
+            xdgRuntimeDir.mkpath("app/org.keepassxc.KeePassXC");
```

Comment:
same, use `socketDir.mkPath({})`. No need to check for existence, the function does that for you and returns true if it exists.